apps/sms/test/marionette/inbox_test.js

```diff
@@ -76,4 +76,75 @@ marionette('Inbox View tests', function() {
       });
     });
   });
+
+  suite('Split Inbox view tests', function() {
+    setup(function() {
+      client.contentScript.inject(__dirname + '/mocks/mock_split_view_mode.js');
+
+      ThreadGenerator.uniqueThreadId = 0;
+
+      var conversations = [];
+      for (var i = 0; i < 10; i++) {
+        conversations.push(
+          ThreadGenerator.generate({
+            participants: ['+1234' + i], numberOfMessages: 5
+          })
+        );
+      }
+
+      storage.setMessagesStorage(
+        conversations, ThreadGenerator.uniqueMessageId
+      );
+
+      messagesApp.launch();
+    });
+
+    test('All conversations are loaded', function() {
+      var inboxView = new InboxView(client);
+
+      client.waitFor(function() {
+        return inboxView.conversations.length === 10;
+      });
+    });
+
+    test('User can enter New Message view from Inbox and go back', function() {
+      var inboxView = new InboxView(client);
+
+      // Make sure we've entered new message view.
+      var newMessageView = inboxView.createNewMessage();
+      newMessageView.assertRecipientsInputFocused();
+
+      // Workaround for bug 1227147.
```

Comment:
looks like it may land really soon, you may be able to remove the workaround before landing